crypto/merkle/hash.go

```diff
@@ -22,5 +22,9 @@ func leafHash(leaf []byte) []byte {
 
 // returns tmhash(0x01 || left || right)
 func innerHash(left []byte, right []byte) []byte {
-	return tmhash.Sum(append(innerPrefix, append(left, right...)...))
+	data := make([]byte, 0, len(innerPrefix)+len(left)+len(right))
+	n := copy(data, innerPrefix)
+	n = copy(data[n:], left)
```

Comment:
Oh oh, this is invalid, it has to be a: `n += copy(data[n:], left)` otherwise you'll just have the length of left

Reply:
Yay, we need to pre-allocate all slice elem 😄